### PR TITLE
Rdm 4190 Case Timeline integrated with Demo app.

### PR DESCRIPTION
**Pull request by andrewfolga**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4190

### Change description ###
Case Timeline integrated with Demo app. High level user documentation. Integration notes to come under: https://tools.hmcts.net/confluence/display/RCCD/Integration+steps

A man web regression around case history aka event log and case history components needed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: demo/src/app/search-filters-consumer.component.html [hunk 1]
```diff
@@ -33,6 +33,7 @@ <h1 class="heading-large">Search Filters component</h1>
                     </ul>
                 </li>
             </ul>
+            <p>Additionally the component saves user selections to local browser storage any time a user click Apply button after selecting dropdown values. This way any time a user is back on the Create Case Filters page the dropdown will be restored from the last applied selection. This can be cleared by clicking Reset button.</p>
```
Comment (andrewfolga) [inline demo/src/app/search-filters-consumer.component.html]: Forgotten docs for search filters consumer.

File: src/shared/components/case-history/case-history.component.ts [hunk 2]
```diff
@@ -32,9 +36,9 @@ export class CaseHistoryComponent implements OnInit {
   ngOnInit() {
     this.caseService.caseView.subscribe(caseDetails => {
       this.caseDetails = caseDetails;
-      this.route.snapshot.paramMap.get(CaseHistoryComponent.PARAM_EVENT_ID);
+      let eventId = this.route.snapshot.paramMap.get(CaseHistoryComponent.PARAM_EVENT_ID) || this.event;
```
Comment (andrewfolga) [inline src/shared/components/case-history/case-history.component.ts]: This has to be retested in man web :(

File: src/shared/components/case-viewer/case-view/case-view.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (andrewfolga) [inline src/shared/components/case-viewer/case-view/case-view.component.spec.ts]: Missing tests from case view extraction.

File: src/shared/components/case-viewer/case-view/case-view.component.ts [hunk 4]
```diff
@@ -31,7 +31,9 @@ export class CaseViewComponent implements OnInit {
           this.caseDetails = caseView;
           this.caseService.announceCase(caseView);
         })
-      ).toPromise();
+      )
+      .toPromise()
```
Comment (andrewfolga) [inline src/shared/components/case-viewer/case-view/case-view.component.ts]: Small refactor to make it consistent with other extracted components error handling.

File: src/shared/components/palette/history/event-log/event-log-table.component.html [hunk 5]
```diff
@@ -10,9 +10,17 @@
   <tbody>
   <tr *ngFor="let event of events" (click)="select(event)" [ngClass]="{'EventLogTable-Selected': selected === event}">
     <td>
-      <a [routerLink]="['./', 'event', event.id, 'history']"
-           [target]="'_blank'"
-           *ngIf="event.state_id !== 'Draft'" class="text-16 event-link">{{event.event_name}}</a>
+      <ng-container [ngSwitch]="isPartOfCaseTimeline">
```
Comment (andrewfolga) [inline src/shared/components/palette/history/event-log/event-log-table.component.html]: Man web retest needed:(

**Comment by adnanakgun** [comment]
Should the cursor here not turn into a pointer ("Case created")?
image

**Comment by andrewfolga** [comment]
Yes noticed that too. Looking at it now.

**Comment by andrewfolga** [comment]
Styling changes applied

**Pull request merged**